### ISSUE TITLE
Register Plugin globally

### ISSUE THREAD

**Issue by motion-work** [ivodolenc/nuxt-gsap-module#2]
Is it possible to register a plugin globally in the ```~/plugins``` folder? for example ScrollTrigger:

```js
const gsap = this.$gsap
const ScrollTrigger = this.$ScrollTrigger

gsap.registerPlugin(ScrollTrigger)
```

**Comment by ivodolenc** [comment]
Hi, sorry for the late reply.

Ok nice suggestion. I did a few tests and it works as expected.

Actually, I will add this as a new `feature` as it will be a nice addition to the module.

The feature will automatically register the plugin globally (after plugin activation in the settings), so you won’t have to worry about it anymore. This will apply to all plugins, not just ScrollTrigger. 

To summarize, you will no longer need to register plugins before use.

Example:

_From this (now)_

```js
export default {
  methods: {
    animateOnScroll() {
      const gsap = this.$gsap
      const ScrollToPlugin = this.$ScrollToPlugin
      const ScrollTrigger = this.$ScrollTrigger

      gsap.registerPlugin(ScrollToPlugin, ScrollTrigger)

      gsap.to(window, { duration: 2, scrollTo: 1000 })

      gsap.to('.box', {
        x: 500,
        ease: 'Power1.easeInOut',
        scrollTrigger: {
          trigger: '.content',
          pin: true,
          end: 'bottom',
          scrub: true,
        },
      })
    },
  },
}
```

_To this (new feature)_

```js
export default {
  methods: {
    animateOnScroll() {
      this.$gsap.to(window, { duration: 2, scrollTo: 1000 })

      this.$gsap.to('.box', {
        x: 500,
        ease: 'Power1.easeInOut',
        scrollTrigger: {
          trigger: '.content',
          pin: true,
          end: 'bottom',
          scrub: true,
        },
      })
    },
  },
}
```

Also, this isn't a breaking change according to the official [docs](https://greensock.com/docs/v3/GSAP/gsap.registerPlugin).

I will update the module docs and add examples.

**Comment by ivodolenc** [comment]
I’ve added a new feature `'register plugin globally after activation'`

- Available in: `1.2.0`
- [Docs](https://github.com/ivodolenc/nuxt-gsap-module#multiple-plugins-usage-example)
- [Example](https://github.com/ivodolenc/nuxt-gsap-module/tree/master/examples/animate-on-scroll)

### Multiple plugins usage example

✅ The module automatically registers plugins globally (after plugin activation in the settings), so you won’t have to worry about it (applies to all plugins).

```js
// nuxt.config.js

{
  gsap: {
    extraPlugins: {
      /**
       * When you enable them, plugins are
       * automatically registered and available globally
       */
      scrollTo: true,
      scrollTrigger: true
    },
    extraEases: {
      expoScaleEase: true
    }
  }
}
```

As official docs say:

> ... There is no harm in registering the same plugin multiple times (but it doesn't help either) ...

So once you activate the plugin in `nuxt.config.js`, there is no need to `re-register` it later in your code. I hope this isn't too confusing. This feature simplifies the usage process and eliminates the need for additional lines of code.

**Comment by ivodolenc** [comment]
Closing this as resolved since it's available in the latest version (`1.2.0`).
